MobileCenterDistribute/MobileCenterDistributeTests/MSDistributeTests.m

```diff
@@ -378,6 +380,143 @@ - (void)testShowConfirmationAlertForMandatoryUpdate {
+//  dispatch_async(dispatch_get_main_queue(), ^{
+//    [expection fulfill];
+//  });
+//  [self waitForExpectationsWithTimeout:1
+//                               handler:^(__attribute__((unused)) NSError *error) {
+//
+//                                 /*
+//                                  * Then
+//                                  */
+//                                 OCMVerify([alertControllerMock alertControllerWithTitle:[OCMArg any]
+//                                                                                 message:details.releaseNotes]);
+//                                 OCMReject(
+//                                           [alertControllerMock addDefaultActionWithTitle:[OCMArg any] handler:[OCMArg any]]);
+//                                 OCMVerify(
+//                                           [alertControllerMock addCancelActionWithTitle:[OCMArg any] handler:[OCMArg any]]);
+//                               }];
+//}
+
+- (void)testDontShowConfirmationAlertIfNoMandatoryReleaseWhileNoNetwork {
+  
+  /*
+   * If
+   */
+  XCTestExpectation *expection = [self expectationWithDescription:@"Confirmation alert has been displayed"];
+  
+  // Mock alert.
+  id alertControllerMock = OCMClassMock([MSAlertController class]);
+  OCMStub([alertControllerMock alertControllerWithTitle:[OCMArg any] message:[OCMArg any]])
+  .andReturn(alertControllerMock);
+  
+  // Mock reachability.
+  id reachabilityMock = OCMClassMock([MS_Reachability class]);
+  OCMStub([reachabilityMock reachabilityForInternetConnection]).andReturn(reachabilityMock);
+  OCMStub([reachabilityMock currentReachabilityStatus]).andDo(^(NSInvocation *invocation) {
+    NetworkStatus test = NotReachable;
+    [invocation setReturnValue:&test];
+  });
+  
+  /*
+   * When
+   */
+  [self.sut checkLatestRelease:@"whateverToken" releaseHash:@"whateverReleaseHash"];
+  dispatch_async(dispatch_get_main_queue(), ^{
+    [expection fulfill];
+  });
+  [self waitForExpectationsWithTimeout:1
+                               handler:^(__attribute__((unused)) NSError *error) {
+                                 
+                                 /*
+                                  * Then
+                                  */
+                                 OCMReject([alertControllerMock alertControllerWithTitle:[OCMArg any]
+                                                                                 message:[OCMArg any]]);
+                                 OCMReject(
+                                           [alertControllerMock addDefaultActionWithTitle:[OCMArg any] handler:[OCMArg any]]);
+                                 OCMReject(
+                                           [alertControllerMock addCancelActionWithTitle:[OCMArg any] handler:[OCMArg any]]);
+                               }];
+}
+
+// FIXME: Somehow this test is failing. NSUserDefaults mock seems not to be working properly.
```

Comment:
I think this is caused by using NSUserDefaults instead of mocking itself. We've observed this issue before and I think we can fix it by mocking NSUserDefaults. Not a merge blocker and we can address the issue soon.